### ISSUE TITLE
Dosbox_x from MS-DOS 6.22: DPMILD32 out of memory

### ISSUE THREAD

**Issue by daerlnaxe** [joncampbell123/dosbox-x#3327]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

hi,
i don't find any solution.

I would to try, under Virtualbox,  to run dosbox_x to see features before to test in real condition and i don't know why but each time i launch dosbox_x , i receive the message "DMILD32: <path> out of memory". I missed probably something.

I tried severla configuration under VirtualBox, because i asked myself if i didn't put a too low hardware, but problem still. 

Thx.

**Comment by rderooy** [comment]
DPMILD32 is a HX DOS extender. I'm fairly certain that it is not meant to run from DOSBox. DOSBox does not provide a HX DOS environment.

**Comment by daerlnaxe** [comment]
Yep but i dl the releaze on the website.. it's theorically easy to run, then if the problem is not only because of the configuration i use to run it, it's better to feedback.

**Comment by Dobby233Liu** [comment]
Maybe try the [LOADLIN version](https://github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.23/dosbox-x-dos-0.83.23-loadlin.zip)?

**Comment by daerlnaxe** [comment]
hi, 

it's a good idea i had too but i have a problem too, as i don't know if it's a good idea to put all i said to myself perhaps it was better stay focus on it.

.bat file doesn't work..

i don't know at all loadlin and the work accomplished here, i looked at the website and tested "Loadlin vmlinuz root=/dev/hda/ "i have a "kernel panic"

**Comment by grapeli** [comment]
The DOS virtual machine must have sufficient memory allocated to run dosbox-x. **128MB should be enough**.
**You don't need to modify any files.**
Just u unpack the downloaded [dosbox-x-dos-0.83.23-loadlin.zip](https://github.com/joncampbell123/dosbox-x/releases/download/dosbox-x-v0.83.23/dosbox-x-dos-0.83.23-loadlin.zip).
And turn on.

Run in a virtual machine [copy.sh](https://copy.sh/v86/) running in the firefox browser. It just works.

https://user-images.githubusercontent.com/452325/158027176-1c4550cf-a88e-40f3-9e3e-f490dc42e4b5.mp4

**Comment by daerlnaxe** [comment]
4Gb  / dual core /  128mb for display under VirtualBox. (4770k oc 16go ddr3 1070 for real machine), i doubt the problem is about it but i had the same idea. this is why the 1st thing i did it's to allow more power to the virtual machine.

No problem under freedos with the same machine, problem still on MS-DOS only i tried on vmware, same...

I looked at your link it's interesting but it's not MS-DOS but FreeDos... It seems you know well how to use this website, if you can try with this packages under MS-DOS i didn't see how to do it, i will answer to the question. 

I even tried PCEM without faith because it's too slow in comparison but it's still to be just MS-Dos so why not just to see if something appears, 533celeron 256Mo, nothing. Exactly the same problems...

**Comment by grapeli** [comment]
It also works from ms-dos 6.22.

https://user-images.githubusercontent.com/452325/158032360-98f263df-6205-49b4-8275-09f5f503c078.mp4

4GB RAM for DOS?

**Comment by grapeli** [comment]
You can run dosbox-x under [copy.sh](https://copy.sh/v86/) without loadlin and dos.
[copy.sh.dosbox-x.zip](https://github.com/joncampbell123/dosbox-x/files/8238311/copy.sh.dosbox-x.zip)

In setup, add these files:
for `Kernel image (bzimage)`: `vmlinuz`
for `initrd`: `dosbox-x.sqf`

and press the `Start Emulation` button.
